src/test/java/edu/garyMooradian/advancedJava/BasicStockServiceTest.java

```diff
@@ -0,0 +1,322 @@
+		 */
+		BigDecimal price = new BigDecimal(5.00);
+		
+		/*
+		 * Verify that dateRecorded is correct for the StockQuote object
+		 * Note: Just trying assertTrue for the hell of it; i.e. instead of assertEquals
+		 */
+		assertTrue("The dateRecorded is not correct", date == quote.getDateRecorded());
+		
+		/*
+		 * Verify that the stockPrice is correct for the StockQuote object
+		 */
+		assertTrue("The stockPrice is not correct", price.equals(quote.getStockPrice()));
+		
+		/*
+		 * Verify the stockSymbol is correct for the stockQuote object
+		 */
+		assertEquals("The stockSymbol is not correct", "OLED", quote.getStockSymbol());
+		
+		
+	}
+	
+	/********************* Tests for getQuoteHistory method *************************/
+	
+	/*
+	 * Testing the GetQuoteHistory method; i.e.
+	 * Testing that the List is populated with two StockQuote objects
+	 * We test for two quotes, not just one, because of the OR condition
+	 * in the while loop; i.e. while(date.before(endDate) || date.equals(endDate))
+	 */
+	@Test
+	public void testGetQuoteHistoryPositive() throws ParseException {
+		BasicStockService basicStockService = new BasicStockService();
+		
+		Calendar calendarStartDate = Calendar.getInstance();
+		calendarStartDate.setTime(new SimpleDateFormat("MM/dd/yyyy").parse("1/20/2019"));
+		
+		Calendar calendarEndDate = Calendar.getInstance();
+		calendarEndDate.setTime(new SimpleDateFormat("MM/dd/yyyy").parse("1/21/2019"));
+		
+		IntervalEnum interval = IntervalEnum.DAILY;
+		
+		/*
+		 * Call the method we are testing
+		 */
+		List<StockQuote> listOfQuotes = basicStockService.getQuoteHistory("OLED", calendarStartDate, calendarEndDate, interval);
+	
+		/*
+		 * Testing the StockQuote for only the stock symbol is good enough since
+		 * we just want to make sure the StockQuote object was created. How well that
+		 * object is created is tested elsewhere; i.e. tested when we tested the getQuote method
+		 */
+		assertEquals("The first stock quote in the list does not contain the proper stock symbol", "OLED", listOfQuotes.get(0).getStockSymbol());
+		assertEquals("The second stock quote in the list does not contain the proper stock symbol", "OLED", listOfQuotes.get(1).getStockSymbol());
+	}
+	
+	
+	/*
+	 * This test passes an end date that is earlier than the start date.
+	 * This should result in no StockQuotes being created, thus the List returned
+	 * should be empty.
+	 */
+	@Test
+	public void testGetQuoteHistoryNegative() throws ParseException {
+		BasicStockService basicStockService = new BasicStockService();
+		
+		Calendar calendarStartDate = Calendar.getInstance();
+		calendarStartDate.setTime(new SimpleDateFormat("MM/dd/yyyy").parse("1/20/2019"));
+		
+		Calendar calendarEndDate = Calendar.getInstance();
+		calendarEndDate.setTime(new SimpleDateFormat("MM/dd/yyyy").parse("1/19/2019"));
+		
+		IntervalEnum interval = IntervalEnum.DAILY;
+		
+		/*
+		 * Call the method we are testing
```

Comment:
could be a single line comment. 
typically only comment lines that aren't obvious
if you find yourself with a lot of lines of code that aren't obvious that indicates a problem with the code itself, change the code so that is clearer rather than just comment.